src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/DefaultVSLSPTagHelperTooltipFactory.cs

```diff
@@ -63,34 +146,42 @@ public override bool TryCreateTooltip(AggregateBoundElementDescription elementDe
                 }
 
                 ClassifyTypeName(runs, descriptionInfo.TagHelperTypeName);
-                TryClassifySummary(runs, descriptionInfo.Documentation);
+
+                if (useContainer)
```

Comment:
Hmmm, given everything doesn't currently support containers I worry we're having that restriction leak too far into the underlying logic. I wonder if you could do an alternate approach where you extract the important info about TagHelpers and then reconstruct the correct runs in the caller? Maybe that'd enable you to remove the container leaking into this method and the classify summary bit below.

Something like:
```C#
private record ElementClassification(IReadOnlyList<VSClassifiedTextRun> Type, IReadOnlyList<VSClassifiedTextRun> Documentation);

....

var tagHelperClassifications = new List<ElementClassification>();
foreach (var descriptionInfo in associatedTagHelperInfos)
{
    var typeNameRuns = new List<VSClassifiedTextRun>();
    ClassifyTypeName(typeNameRuns, descriptionInfo.TagHelperTypeName);

    var summaryRuns = new List<VSClassifiedTextRun>();
    TryClassifySummary(summaryRuns, descriptionInfo.Documentation);

    // This could be your "out" type or something equivalent and then the caller could iterate through each classification and be the "orchestrator" for how to fit each of the TagHelpers runs into a coherent  structure
    tagHelperClassifications.Add(new(typeNameRuns, summaryRuns));
}
```


Ditto with the attribute classification bit

Reply:
Good call! Just updated.